kmk/modules/trilayerstate.py

```diff
@@ -0,0 +1,211 @@
+'''One layer isn't enough. Adds keys to get to more of them'''
+from kmk.keys import KC, make_argumented_key, make_key
+from kmk.modules.holdtap import HoldTap, HoldTapKeyMeta
+from kmk.utils import Debug
+import time
+debug = Debug(__name__)
+
+tri = '0'
+
+
+def layer_key_validator(layer, kc=None):
+    '''
+    Validates the syntax (but not semantics) of a layer key call.  We won't
+    have access to the keymap here, so we can't verify much of anything useful
+    here (like whether the target layer actually exists). The spirit of this
+    existing is mostly that Python will catch extraneous args/kwargs and error
+    out.
+    '''
+    return LayerKeyMeta(layer, kc)
+
+
+def layer_key_validator_lt(layer, kc, prefer_hold=False, **kwargs):
+    return HoldTapKeyMeta(tap=kc, hold=KC.MO(layer), prefer_hold=prefer_hold, **kwargs)
+
+
+def layer_key_validator_tt(layer, prefer_hold=True, **kwargs):
+    return HoldTapKeyMeta(
+        tap=KC.TG(layer), hold=KC.MO(layer), prefer_hold=prefer_hold, **kwargs
+    )
+
+
+class LayerKeyMeta:
+    def __init__(self, layer, kc=None):
+        self.layer = layer
+        self.kc = kc
+
+
+class Layers(HoldTap):

```

Comment:
Would it make more sense to make this subclass the existing layers module, sort of like how ModTap subclasses HoldTap?

Reply:
I was thinking the same thing. That's sort of how I built it to begin with. I'll update tomorrow

Reply:
> Would it make more sense to make this subclass the existing layers module, sort of like how ModTap subclasses HoldTap?

Alright. I added as a subclass of layer. You're right, it does make more sense.